src/modules/admin/components/Profile/ProfileEdit.tsx

```diff
@@ -78,12 +99,19 @@ const ProfileEdit = ({ colony }: Props) => {
     guideline,
     website,
   } = colony;
-  const transform = useCallback(
-    pipe(
-      withKey(colonyAddress),
-      mergePayload({ colonyAddress }),
-    ),
-    [colonyAddress],
+
+  const [editColony] = useMutation(EditColonyProfileDocument);
```

Comment:
There should be a hook in the generated file (`useEditColonyProfileMutation`, I think) which you can immediately use. Furthermore you can define the parameters directly in the mutation and you don't need to use `useCallback`

Reply:
This won't work in this case, as you still need the values from the form. _(ie: `profile` in this case)_

And as far as I can see, you can't specify a custom method to `useEditColonyProfileMutation`, just the values to be passed into the mutation

This is the exact same way you are editing the user profile:

https://github.com/JoinColony/colonyDapp/blob/712d0d025cba3a5095f14b32961f267690396c80/src/modules/users/components/UserProfileEdit/UserProfileEdit.tsx#L79-L83

But maybe I'm missing something... ?

Reply:
Oh yeah, sorry you're right, scratch the last part, you'll need `useCallback`. But you _can_ use the mutation hook (and I should, too, I just forgot about it in the edit user case)

Reply:
I updated it to use the generated code. See here: https://github.com/JoinColony/colonyDapp/blob/feature/user-metadata-mutations/src/modules/users/components/UserProfileEdit/UserProfileEdit.tsx#L77

Reply:
Yup, just saw your commits.

I've updated as well.